packages/amazonq/src/extension.ts

```diff
@@ -4,12 +4,78 @@
  */
 
 import * as vscode from 'vscode'
-import { activateShared, deactivateShared } from './extensionShared'
+import { activateAmazonQCommon, amazonQContextPrefix, deactivateCommon } from './extensionCommon'
+import { DefaultAmazonQAppInitContext, activate as activateCWChat } from 'aws-core-vscode/amazonq'
+import { activate as activateQGumby } from 'aws-core-vscode/amazonqGumby'
+import { ExtContext } from 'aws-core-vscode/shared'
+import { updateDevMode } from 'aws-core-vscode/dev'
+import { CommonAuthViewProvider } from 'aws-core-vscode/login'
+import { isExtensionActive, VSCODE_EXTENSION_ID } from 'aws-core-vscode/utils'
+import { registerSubmitFeedback } from 'aws-core-vscode/feedback'
+import { DevFunction } from 'aws-core-vscode/dev'
 
 export async function activate(context: vscode.ExtensionContext) {
-    await activateShared(context, false)
+    // IMPORTANT: No other code should be added to this function. Place it in one of the following 2 functions where appropriate.
+    await activateAmazonQCommon(context, false)
+    await activateAmazonQNonCommon(context)
+}
+
+/**
+ * The code in this function is not common, implying it only works in Node.js and not web.
+ * The goal should be for this to not exist and that all code is "common". So if possible make
+ * the code compatible with web and move it to {@link activateAmazonQCommon}.
+ */
+async function activateAmazonQNonCommon(context: vscode.ExtensionContext) {
```

Comment:
"NonCommon" means "node only" right? "Nodejs" is probably a less awkward name
```suggestion
async function activateAmazonQNodejs(context: vscode.ExtensionContext) {
```